homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -80,6 +88,47 @@ async def _async_init_client(self) -> None:
                 name="lm_websocket_task",
             )
 
+        # initialize Bluetooth
+        if self.config_entry.data.get(CONF_USE_BLUETOOTH, True):
+            username = self.config_entry.data[CONF_USERNAME]
+            if (mac_address := self.config_entry.data.get(CONF_MAC, "")) and (
+                name := self.config_entry.data.get(CONF_NAME, "")
+            ):
+                # coming from discovery or last time we used bluetooth
+                _LOGGER.debug("Initializing with known Bluetooth device")
+                await self.lm.init_bluetooth_with_known_device(
+                    username, mac_address, name
+                )
+                self._use_bluetooth = True
+            else:
+                # check if there are any bluetooth adapters to use
+                count = bluetooth.async_scanner_count(self.hass, connectable=True)
+                if count > 0:
+                    _LOGGER.debug(
+                        "Found Bluetooth adapters, initializing with Bluetooth"
+                    )
+
+                    try:
+                        await self.lm.init_bluetooth(
+                            username=username,
+                            init_client=False,
+                            bluetooth_scanner=bluetooth.async_get_scanner(self.hass),
```

Comment:
Why pass the scanner here?

If you already know the mac you could connect or fetch the service info with:
https://developers.home-assistant.io/docs/core/bluetooth/api?_highlight=blueto#fetching-the-latest-bluetoothserviceinfobleak-for-a-device

Or if you want all the discoveries https://developers.home-assistant.io/docs/core/bluetooth/api?_highlight=blueto#fetching-all-discovered-devices

If you pass the scanner and do a scan its not actually going to do anything besides wait and than return the same data

Reply:
This is the case where I do not know that mac, but I simply didn't know about this endpoint. Will change it.

Reply:
will the `async_discovered_service_info` return any BT device HA sees, or only the devices which are eligible for this integration (aka defined in my manifest)

Reply:
It will return everything HA sees. You'll need to filter

Look in the switchbot or govee-ble flows to see how its done there

Reply:
done